### PR TITLE
8 dao crud office

### PR DESCRIPTION
**Pull request by jadatix**
Fixes #8

### CODE REVIEW

File: src/main/java/org/jadatix/carbooking/dao/OfficeDao.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    public Optional<Office> get(Long id) {
+        Office res = entityManager.find(Office.class, id);
+        return Optional.ofNullable(res);
+    }
+
+    public List<Office> getAll() {
+        String query = new StringBuilder().append("from ").append(tableName).toString();
+        Query theQuery = (Query) entityManager.createQuery(query, Office.class);
```
Comment (seshotake) [inline src/main/java/org/jadatix/carbooking/dao/OfficeDao.java:30]: Why not use just strings concatenation?
```suggestion
        Query theQuery = (Query) entityManager.createQuery("from " + tableName, Office.class);
```

File: src/main/java/org/jadatix/carbooking/api/v1/offices/OfficeController.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    public ResponseEntity<List<Office>> getAllOffices() {
+        List<Office> offices = officeService.getAllOffices();
+        HttpStatus status = offices != null ? HttpStatus.OK : HttpStatus.INTERNAL_SERVER_ERROR;
+        return new ResponseEntity<>(offices, status);
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<Office> getOfficeById(@PathVariable Long id) {
+        Office responseOffice = officeService.getOfficeById(id).get();
+        HttpStatus status = responseOffice != null ? HttpStatus.OK : HttpStatus.BAD_REQUEST;
```
Comment (seshotake) [inline src/main/java/org/jadatix/carbooking/api/v1/offices/OfficeController.java:39]: Isn't that always the true?

**Review by seshotake — CHANGES_REQUESTED** [review]

File: src/main/java/org/jadatix/carbooking/api/v1/offices/OfficeController.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        return new ResponseEntity<>(responseOffice, status);
+    }
+
+    @PutMapping("/{id}")
+    public ResponseEntity<Office> updateOffice(@PathVariable Long id, @RequestBody Office office) {
+        office.setId(id);
+        officeService.updateOffice(office);
+        return new ResponseEntity<>(null, HttpStatus.OK);
```
Comment (seshotake) [inline src/main/java/org/jadatix/carbooking/api/v1/offices/OfficeController.java:47]: Should it return office?
```suggestion
    public void updateOffice(@PathVariable Long id, @RequestBody Office office) {
        officeService.updateOffice(office);
```

File: src/main/java/org/jadatix/carbooking/api/v1/offices/OfficeController.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+    @DeleteMapping("/{id}")
+    public ResponseEntity<?> deleteOfficeById(@PathVariable Long id) {
+        try {
+            officeService.deleteOfficeById(id);
+            return new ResponseEntity<>(null, HttpStatus.OK);
+        } catch (RuntimeException exception) {
+            return new ResponseEntity<>(exception.getMessage(), HttpStatus.BAD_REQUEST);
+        }
+
```
Comment (seshotake) [inline src/main/java/org/jadatix/carbooking/api/v1/offices/OfficeController.java:58]: Maybe, just use?

```suggestion
officeService.deleteOfficeById(id);
```

File: src/main/java/org/jadatix/carbooking/dao/OfficeDao.java (hunk 1 above)
Comment (seshotake) [inline src/main/java/org/jadatix/carbooking/dao/OfficeDao.java:30]: is it necessary cast `TypedQuery<T>` to `Query`?

https://docs.oracle.com/javaee/7/api/javax/persistence/EntityManager.html#createQuery-java.lang.String-java.lang.Class-

If you don't need `TypedQuery` just use `createNativeQuery`

https://docs.oracle.com/javaee/7/api/javax/persistence/EntityManager.html#createNativeQuery-java.lang.String-

**Review by seshotake — CHANGES_REQUESTED** [review]

File: src/main/java/org/jadatix/carbooking/dao/OfficeDao.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+
+import javax.persistence.EntityManager;
+import java.util.List;
+import java.util.Optional;
+
+@Repository
+public class OfficeDao {
```
Comment (voltor) [inline src/main/java/org/jadatix/carbooking/dao/OfficeDao.java:14]: DAO isn't a Repository, here better to use `@Service` or `@Component`

File: src/main/java/org/jadatix/carbooking/dao/OfficeDao.java [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    private EntityManager entityManager;
```
Comment (voltor) [inline src/main/java/org/jadatix/carbooking/dao/OfficeDao.java:15]: Could you use here a Repository; more see SpringData

**Pull request merged**